src/test/java/org/broadinstitute/hellbender/utils/SATagBuilderUnitTests.java

```diff
@@ -0,0 +1,183 @@
+package org.broadinstitute.hellbender.utils;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SATagBuilderUnitTests {
+
+    @Test
+    public static void testSetReadsAsSupplementalNo() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        GATKRead primarySup = ArtificialReadUtils.createArtificialRead(header, "read1", 1, 10000, new byte[] {}, new byte[] {}, "4M");
+        primarySup.setMappingQuality(100);
+        primarySup.setAttribute("NM",20);
+        primarySup.setIsReverseStrand(true);
+        GATKRead secondarySup = ArtificialReadUtils.createArtificialRead(header, "read2", 2, 10001, new byte[] {}, new byte[] {}, "4S");
```

Comment:
Can you name all of the supplementary reads the same name since that is what we would typically expect?